include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -977,7 +977,8 @@ namespace internal
           cell_vectorization_categories_strict || hp_functionality_enabled;
         unsigned int dofs_per_cell = 0;
         for (const auto &info : dof_info)
-          dofs_per_cell = std::max(dofs_per_cell, info.dofs_per_cell[0]);
+          for (const auto &dofs : info.dofs_per_cell)
+            dofs_per_cell = std::max(dofs_per_cell, dofs);
```

Comment:
This looks reasonable. We should rename  the variable `dofs_per_cell` to `max_dofs_per_cell` and add an assert in `TaskInfo::create_blocks_serial` ruling out 0. Maybe we could add a comment here why we are taking here the maximum: it seems to be a useful heuristic for a good blocking!?